jupyterhub/spawner.py

```diff
@@ -1468,15 +1468,22 @@ def get_env(self):
             self.clear_state()
             return 0
 
-        # send signal 0 to check if PID exists
-        # this doesn't work on Windows, but that's okay because we don't support Windows.
-        alive = await self._signal(0)
+        alive = await self._is_single_user_process_alive()
         if not alive:
             self.clear_state()
             return 0
         else:
             return None
 
+    async def _is_single_user_process_alive(self):
```

Comment:
Since this is intended to be overridable it's part of the public API, so I think the leading `_` could be removed?

Reply:
It's intended to be overridable by subclasses indeed. The way I see it's intended not to be called by outside code, so I think it's not part of the public API per se. Making it part of that is not needed for making WinLocalProcessSpawner work. But if you guys think it could be useful for other code, why not.

Reply:
Of course, like a public API method, it has the constraint that once we merge it in and WinLocalProcessSpawner reimplements it, there's an involved refactoring process to change the name, for example. Even if it's not meant to be called by outside code.

Reply:
I have removed this method in my latest commit, it's not needed, pls see below.